sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1044,7 +1044,7 @@ object EliminateSorts extends Rule[LogicalPlan] {
 
   private def isOrderIrrelevantAggs(aggs: Seq[NamedExpression]): Boolean = {
     def isOrderIrrelevantAggFunction(func: AggregateFunction): Boolean = func match {
-      case _: Min | _: Max | _: Count => true
+      case _: Min | _: Max | _: Count | _: BitAggregate | _: HyperLogLogPlusPlus => true
```

Comment:
`HyperLogLogPlusPlus` is an estimation, will the input order affect the estimation?

Reply:
Internally it uses the hash values. From the hash it calculates the bucket id and the max number of leading zeros for all the inputs for each buckets. All of these are deterministic operations, that do not depend on the order of insertion.

Reply:
(IMHO) even if so, we don't have any restriction about the deteminisity of the `HyperLogLogPlusPlus` impmentation. So, adding it in this rule  look a bit dangerous.

Reply:
I see your point. I'll remove that then, but keep the `BitAggregate`